Babylon/groups/api/groups/solution/commands/get.py

```diff
@@ -1,42 +1,84 @@
-import logging
+import json
+from logging import getLogger
 from pprint import pformat
 from typing import Optional
-import json
 
-import cosmotech_api
+from click import argument
 from click import command
 from click import make_pass_decorator
 from click import option
+from click import Path
 from cosmotech_api.api.solution_api import SolutionApi
-from ......utils.api import convert_keys_case, underscore_to_camel
+from cosmotech_api.exceptions import NotFoundException
+from cosmotech_api.exceptions import UnauthorizedException
 
+from ......utils.api import convert_keys_case
+from ......utils.api import filter_api_response_item
+from ......utils.api import underscore_to_camel
+from ......utils.decorators import allow_dry_run
 from ......utils.decorators import require_deployment_key
 from ......utils.decorators import timing_decorator
 
-logger = logging.getLogger("Babylon")
+logger = getLogger("Babylon")
 
 pass_solution_api = make_pass_decorator(SolutionApi)
 
 
 @command()
+@allow_dry_run
 @pass_solution_api
-@require_deployment_key("organization_id", "organization_id")
-@require_deployment_key("solution_id", "solution_id")
-@option("-o", "--output_file", "output_file", help="File to which content should be outputed (json-formatted)")
 @timing_decorator
-def get(solution_api: SolutionApi, solution_id: str, organization_id: str, output_file: Optional[str] = None):
-    """Get the state of the solution in the API"""
+@argument("solution_id")
```

Comment:
Add input file option to get ID from a file

Reply:
I don't think that it is necessary.
If someone creates a solution for example if he wants to see the created solution details he will just open the output file generated by create commands.

But creating a new file, copying the id, setting the file format ... and then running the command is painful I think.
it's easier to get the list of all solutions, copy the id of the one which I'm interested in and run the command with.

Also `get` is not a part an deployment it's just an accessory commands

Reply:
There could be cases where the solution already exists and we want to read it to act on other parts of the infrastructure. Babylon is not just for deployment